src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/AmendmentPreviousNameToEdifactMapper.java

```diff
@@ -0,0 +1,39 @@
+package uk.nhs.digital.nhsconnect.nhais.translator.amendment;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.PersonPreviousName;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AmendmentPreviousNameToEdifactMapper extends AmendmentToEdifactMapper {
+
+    @Override
+    protected Optional<Segment> mapPatches(JsonPatches patches) {
+        if (shouldCreatePersonPreviousNameSegment(patches)) {
+            var personPreviousName = PersonPreviousName.builder()
+                .previousFamilyName(patches.getPreviousSurname().map(this::getValue).orElse(null))
+                .build();
+
+            return Optional.of(personPreviousName);
+        }
+        return Optional.empty();
+    }
+
+    private boolean shouldCreatePersonPreviousNameSegment(JsonPatches patches) {
+        return patches.getPreviousSurname().isPresent();
+    }
```

Comment:
Is this method `shouldCreatePersonPreviousNameSegment()` needed?
````java
protected Optional<Segment> mapPatches(JsonPatches patches) {
	return patches.getPreviousSurname()
		.map(this::getValue)
		.map(value -> PersonPreviousName.builder().previousFamilyName(value).build());
}
````
or
````java
protected Optional<Segment> mapPatches(JsonPatches patches) {
	return patches.getPreviousSurname()
		.map(this::getValue)
		.map(PersonPreviousName.builder()::previousFamilyName)
        .map(PersonPreviousName.PersonPreviousNameBuilder::build)
}
````

Reply:
I refactored this a bit. Pls review